web/views.py

```diff
@@ -40,7 +42,29 @@ def project_detail(request,project_id):
 	project = get_object_or_404(Project, project_id=project_id)
 	return render(request, 'pages/project_detail.html', {'project': project})
 
+def project_new(request):
+	if request.method == "POST":
+		projectform = ProjectForm (request.POST)
+		if projectform.is_valid():
+			project = projectform.save()
+			return redirect('web.views.project_detail', project_id=project.project_id)
+	else:
+		projectform = ProjectForm()
+
+	return render(request, 'pages/project_edit.html', {'projectform': projectform})
+
+def project_edit(request, project_id):
+	project = get_object_or_404(Project, project_id=project_id)
```

Comment:
Ok. What I'm talking about is all about 404 error handling.
@ercchy wrote: 
"It is much better to get an empty dict (which the model will return if there will be not projects) and test for that in the template than this."
If testing on non existing project for example http://127.0.0.1:8000/project/11/edit/ It's raising an error (DoesNotExist at /project/11/edit/  Project matching query does not exist.) and not an empty dict. Probably I'm missing the point what Erika meant to said.